drivers/pse/si3474/si3474_setup.c

```diff
@@ -0,0 +1,141 @@
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+#include "si3474.h"
+#include "si3474_reg.h"
+
+LOG_MODULE_DECLARE(SI3474, CONFIG_PSE_LOG_LEVEL);
+
+static void si3474_gpio_callback(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct si3474_data *drv_data = CONTAINER_OF(cb, struct si3474_data, gpio_cb);
+	const struct si3474_config *cfg = drv_data->dev->config;
+
+	ARG_UNUSED(pins);
+
+	gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_DISABLE);
+
+	k_sem_give(&drv_data->gpio_sem);
+}
+
+int si3474_event_trigger_set(const struct device *dev, pse_event_trigger_handler_t handler)
+{
+	struct si3474_data *drv_data = dev->data;
+	const struct si3474_config *cfg = dev->config;
+	int res = 0;
+
+	gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_DISABLE);
+
+	drv_data->event_ready_handler = handler;
+
+	gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_EDGE_TO_INACTIVE);
+
+	return res;
+}
+
+static void si3474_thread_cb(const struct device *dev)
+{
+	struct si3474_data *drv_data = dev->data;
+	const struct si3474_config *cfg = dev->config;
+
+	if (drv_data->event_ready_handler != NULL) {
+		drv_data->event_ready_handler(dev);
+	}
+
+	gpio_pin_interrupt_configure_dt(&cfg->gpio_int, GPIO_INT_EDGE_TO_INACTIVE);
+}
+
+static void si3474_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+
+	struct si3474_data *drv_data = p1;
+
+	while (1) {
+		k_sem_take(&drv_data->gpio_sem, K_FOREVER);
+		si3474_thread_cb(drv_data->dev);
+	}
+}
+
+int si3474_init_interrupt(const struct device *dev)
+{
+	struct si3474_data *drv_data = dev->data;
+	const struct si3474_config *cfg = dev->config;
+	int result = 0;
```

Comment:
_res_ is used all over the place to carry the result. Please be consistent.
Rename result to res.